test/framework/src/main/java/org/elasticsearch/test/ESTestCase.java

```diff
@@ -2211,6 +2211,10 @@ protected static SecureRandom secureRandomFips(final byte[] seed) throws NoSuchA
      */
     public static final TimeValue SAFE_AWAIT_TIMEOUT = TimeValue.timeValueSeconds(10);
 
+    /**
+     * Await on the given {@link CyclicBarrier} with a timeout of {@link #SAFE_AWAIT_TIMEOUT}, preserving the interrupt flag and converting
```

Comment:
```suggestion
     * Await on the given {@link CyclicBarrier} with a timeout of {@link #SAFE_AWAIT_TIMEOUT}, preserving the thread interrupt flag and converting
```

Reply:
I would add "thread interrupt flag" everywhere to clarify it a little -- I read about this java behavior a little while ago, but otherwise I would have been a little mystified.

Reply:
++ (I checked, and technically it's called the thread's _interrupt status flag_, so I used that terminology)